jwst/straylight/straylight.py

```diff
@@ -229,3 +230,67 @@ def correct_xartifact(input_model, modelpars):
 
     log.info("Cross-artifact model complete.")
     return output
+
+def clean_showers(self, input_model, allregions):
+    """
+    Corrects the MIRI MRS data for straylight produced by residual cosmic ray showers.
+
+    Parameters
+    ----------
+    input_model : `~jwst.datamodels.IFUImageModel`
+        Science data to be corrected.
+
+    allregions : FITS array
+        Holds the regions information mapping MRS pixels to slices (3-D, planes for different throughput)
+
+    Returns
+    -------
+    output : `~jwst.datamodels.IFUImageModel`
+        Straylight-subtracted science data.
+
+    """
+
+    log.info("Applying correction for residual cosmic ray showers.")
+
+    plane = self.shower_plane
+    low_reject = self.shower_low_reject
+    high_reject = self.shower_high_reject
+    x_stddev = self.shower_x_stddev
+    y_stddev = self.shower_y_stddev
+
+    # Create a copy of the input data array that will be modified
+    # for use in the straylight calculations
+    usedata = input_model.data.copy()
+    mask_dq = input_model.dq
+
+    # Which throughput plane of the slice map should be used?
+    regions = allregions[plane,:,:]
+
+    # NaN-out the science pixels by using the slice footprint regions
+    scipix = np.where(regions != 0)
+    usedata[scipix] = np.nan
+
+    # NaN-out pixels that should not be used for computation
+    all_flags = (dqflags.pixel['DO_NOT_USE'] + dqflags.pixel['REFERENCE_PIXEL'])
+    # where are pixels set to any one of the all_flags cases
+    testflags = np.bitwise_and(mask_dq, all_flags)
+    # where are testflags ne 0 and mask == 1
+    bad_flags = np.where(testflags != 0)
+    usedata[bad_flags] = np.nan
```

Comment:
The bitwise-or and -and operators and boolean indexing are slightly preferable here, for readability and performance.
 ```suggestion
    # NaN-out pixels that should not be used for computation
    all_flags = (dqflags.pixel['DO_NOT_USE'] | dqflags.pixel['REFERENCE_PIXEL'])
    # where are pixels set to any one of the all_flags cases
    testflags = mask_dq & all_flags
    # where are testflags ne 0 and mask == 1
    bad_flags = (testflags != 0)
    usedata[bad_flags] = np.nan
```